flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/nodes/common/CommonLookupJoin.scala

```diff
@@ -419,19 +419,17 @@ abstract class CommonLookupJoin(
       joinKeyPairs: Array[IntPair],
       joinInfo: JoinInfo,
       allLookupKeys: Map[Int, LookupKey]): Option[RexNode] = {
-    val remainingPairs = joinKeyPairs.filter(p => !checkedLookupFields.contains(p.target))
+    val joinPairs = joinInfo.pairs().asScala.toArray
+    val remainingPairs = joinPairs.filter(p => !checkedLookupFields.contains(p.target))
```

Comment:
This is still not right. 

If we change the new added test query to the following:

```sql 
val sql = "SELECT T.id, T.content, D.age  FROM T JOIN userTable " +
      "for system_time as of T.proctime AS D ON T.content = concat(D.name, '!') AND D.age = 11"
```

Just remove the `D.name` from SELECT, the result will be wrong. 

`checkedLookupFields` is the field index of the dim table, we should reflect to the index of calc.

The following diagram explains the internals. 

```java
  *      Join (l.name = r.name)
  *    /     \
  * RelNode  Calc (concat(name, "!") as name)
  *           |
  *        DimTable (lookup: age=11)
  *     (age, id, name)
  *
  *
  * joinPairs= "0=0" (equal condition of Join)
  * joinKeyPairs = empty (left input field index to dim table field index)
  * checkedLookupFields = 0 (lookup index of "age" field in table)
```

Reply:
Thank you. This help me a lot. I have fixed.